### PR TITLE
Update GoG script to support multi setup

### PR DESCRIPTION
**Pull request by qparis**

### CODE REVIEW

File: Engines/Wine/QuickScript/GoG Script/script.js [hunk 1]
```diff
@@ -12,15 +12,17 @@ GogScript.prototype = Object.create(QuickScript.prototype);
 GogScript.prototype.constructor = GogScript;
 
 /**
- * Sets the setup file name so that the script can fetch it from gog.com
- * @param {string} setupFileName The setup file name
+ * Sets the setup(s file(s) name so that the script can fetch it from gog.com
+ * @param {string|string[]} setupFileName The setup file name
```
Comment (plata) [inline Engines/Wine/QuickScript/GoG Script/script.js]: "name(s)"

File: Engines/Wine/QuickScript/GoG Script/script.js [hunk 2]
```diff
@@ -34,20 +36,47 @@ GogScript.prototype.loginToGog = function (setupWizard) {
...
+        .get();
+}
+
+/**
+ * Download the setup resources in the same directory, and returns the path of the .exe
+ * @param setupWizard
+ * @returns {*}
+ */
+GogScript.prototype.download = function (setupWizard) {
+    var _setupDirectory = createTempDir();
```
Comment (plata) [inline Engines/Wine/QuickScript/GoG Script/script.js]: why `_`? It's only a local variable.

File: Engines/Wine/QuickScript/GoG Script/script.js [hunk 3]
```diff
@@ -34,20 +36,47 @@ GogScript.prototype.loginToGog = function (setupWizard) {
...
+    var that = this;
+    if (Array.isArray(this._setupFileName)) {
+        var foundExecutable = null;
+        this._setupFileName.forEach(function(setupFileName) {
```
Comment (plata) [inline Engines/Wine/QuickScript/GoG Script/script.js]: Naming is confusing: `_setupFileName` can be an array of setup file names.

File: Engines/Wine/QuickScript/GoG Script/script.js [hunk 4]
```diff
@@ -56,7 +85,7 @@ GogScript.prototype.go = function () {
     setupWizard.presentation(this._name, this._editor, this._applicationHomepage, this._author);
 
     this.loginToGog(setupWizard);
-    this.download(setupWizard);
+    var _setupFile = this.download(setupWizard);
```
Comment (plata) [inline Engines/Wine/QuickScript/GoG Script/script.js]: why `_`?

File: Engines/Wine/QuickScript/GoG Script/script.js [hunk 5]
```diff
@@ -12,15 +12,17 @@ GogScript.prototype = Object.create(QuickScript.prototype);
...
+ * Sets the setup(s file(s) name so that the script can fetch it from gog.com
+ * @param {string|string[]} setupFileName The setup file name
  * @returns {GogScript} This
  */
 GogScript.prototype.gogSetupFileName = function (setupFileName) {
     this._setupFileName = setupFileName;
     return this;
 }
 
+GogScript.prototype.gogSetupFileNames = GogScript.prototype.gogSetupFileName
```
Comment (plata) [inline Engines/Wine/QuickScript/GoG Script/script.js]: Better:
```Javascript
this._setupFileNames = [setupFileName];
```

And then be consistent.

File: Engines/Wine/QuickScript/GoG Script/script.js [hunk 6]
```diff
@@ -12,15 +12,17 @@ GogScript.prototype = Object.create(QuickScript.prototype);
 GogScript.prototype.constructor = GogScript;
 
 /**
- * Sets the setup file name so that the script can fetch it from gog.com
- * @param {string} setupFileName The setup file name
+ * Sets the setup(s file(s) name so that the script can fetch it from gog.com
```
[the remark this answers is not in the document]
Comment (qparis) [reply Engines/Wine/QuickScript/GoG Script/script.js]: Done

File: Engines/Wine/QuickScript/GoG Script/script.js (hunk 1 above)
Comment (qparis) [reply Engines/Wine/QuickScript/GoG Script/script.js]: Done

File: Engines/Wine/QuickScript/GoG Script/script.js (hunk 5 above)
Comment (qparis) [reply Engines/Wine/QuickScript/GoG Script/script.js]: Done

File: Engines/Wine/QuickScript/GoG Script/script.js (hunk 2 above)
Comment (qparis) [reply Engines/Wine/QuickScript/GoG Script/script.js]: Done

File: Engines/Wine/QuickScript/GoG Script/script.js (hunk 4 above)
Comment (qparis) [reply Engines/Wine/QuickScript/GoG Script/script.js]: Done

File: Engines/Wine/QuickScript/GoG Script/script.js (hunk 3 above)
Comment (qparis) [reply Engines/Wine/QuickScript/GoG Script/script.js]: Done

File: Engines/Wine/QuickScript/GoG Script/script.js [hunk 7]
```diff
@@ -12,12 +12,22 @@ GogScript.prototype = Object.create(QuickScript.prototype);
 GogScript.prototype.constructor = GogScript;
 
 /**
- * Sets the setup file name so that the script can fetch it from gog.com
- * @param {string} setupFileName The setup file name
+ * Sets one setup file name so that the script can fetch it from gog.com
+ * @param {string|string[]} setupFileNames The setup file name(s)
```
Comment (plata) [inline Engines/Wine/QuickScript/GoG Script/script.js]: Comment doesn't match. Should be:
```Javascript
@param {string} setupFileName The setup file name
```

File: Engines/Wine/QuickScript/GoG Script/script.js [hunk 8]
```diff
@@ -12,12 +12,22 @@ GogScript.prototype = Object.create(QuickScript.prototype);
...
  */
 GogScript.prototype.gogSetupFileName = function (setupFileName) {
-    this._setupFileName = setupFileName;
+    this._setupFileNames = [setupFileName];
+    return this;
+}
+
+/**
+ * Sets the setup file(s) name so that the script can fetch it from gog.com
+ * @param {string|string[]} setupFileNames The setup file name(s)
```
Comment (plata) [inline Engines/Wine/QuickScript/GoG Script/script.js]: Comment doesn't match. Should be:
```Javascript
@param {string[]} setupFileNames The setup file names
```

**Comment by plata** [comment]
Please fix the Codacy issues before merging. You can ignore the GitHub pages build in Travis.

**Pull request merged**